_posts/2017-04-04-what-every-developer-should-know-about-ir.md

```diff
@@ -0,0 +1,250 @@
+
+```
+(dog OR hound OR mongrel OR pooch OR stray)
+AND
+(diet OR nourishment OR nutrition OR sustenance)
+```
+
+Could some of the documents talk about *hounds* or *pooches* (in plural form) and not mention singular form at all? It's possible, hence:
+
+```
+(dog OR dogs OR hound OR hounds OR mongrel OR mongrels
+  OR pooch OR pooches OR stray)
+AND
+(diet OR nourishment OR nutrition OR sustenance)
+```
+
+Or we might even utilize wildcard operator:
+
+```
+(dog* OR hound* OR mongrel* OR pooch* OR stray*)
+AND
+(diet* OR nourishment* OR nutrition* OR sustenance*)
+```
+
+We begin to see, that we should be really good with words or even have a dictionary with us when we craft our queries. Further on we realise, that making queries like this, is a job for a professional information retrieval specialist. Building those OR-facets seems a horrible task and practically no system is anymore implemented like this. Yet, some of the largest and dominant systems still use boolean model as a main interface - it's fair to say, that these types of systems are meant for expert usage.
+
+There has been loads of systems that use synonym dictionaries to add synonyms automatically to queries, so user don't have to think the synonyms at all. This is called *query expansion*, which is very important technique when helping user with queries. The problem is that you need a really good synonym dataset to make something like this automatically. This is usually really beneficial in large collections that deal with wide range of subjects - such as news collections - and is not that beneficial in smaller rather heavily specialised collections where users usually know the terms documents contain. Query expansion is not limited to boolean model, but usable in various other retrieval models as well, which we'll take a peek next.
+
+## Vector space model
+
+Vector space model is nowadays the most predominant retrieval model. I try not to get too mathematical, but in vector space model index is defined as a vector space as the name suggest. Now, this might seem like a total **mumbo-jumbo**, but one can think vector space as documents being vectors in space in which terms are dimensions (yes, there really is quite a lot of dimensions...) and weight of terms is "the juice" how far these vectors reach. Because vector calculus contains some very intriguing methods for calculating similarities we take advantage of that, and actually calculate the weighted similarity between query and a document and order result set with that. This is called *partial matching*, because documents that do not contain all the terms in query can easily be included to the result set. Interested reader is suggested to revise through suggested reading below, as we will not address the mathematical basis more here.
```

Comment:
as the name suggest_s_
ehkä?  vector space as the documents being vectors in _a_ space in which _the_ terms are _the_ dimensions
there really _are_ quite a lot of dimensions
and _the_ weight of _the_ terms 
similarity between _a_ query and a document and order _the_ result set
because _the_ documents that do not 
suggest toistuu, oisko vaikka näin? _The_ interested reader is encouraged to revise through _the_ suggested reading below